rapid_response_xblock/block.py

```diff
@@ -153,31 +155,45 @@ def responses(self, request=None, suffix=None):  # pylint: disable=unused-argume
         """
         Returns student responses for rapid-response-enabled block
         """
-        status = RapidResponseBlockStatus.objects.filter(
-            problem_usage_key=self.wrapped_block_usage_key,
-            course_key=self.course_key
-        ).first()
-        is_open = False if not status else status.open
-        response_data = RapidResponseSubmission.objects.filter(
+        runs = RapidResponseRun.objects.filter(
             problem_usage_key=self.wrapped_block_usage_key,
             course_key=self.course_key,
-        ).values('answer_id').annotate(count=Count('answer_id'))
-        response_counts = {item['answer_id']: item['count'] for item in response_data}
+        ).order_by('created')
+        is_open = runs.filter(open=True).exists()
+        response_data = RapidResponseSubmission.objects.filter(
+            run__problem_usage_key=self.wrapped_block_usage_key,
+            run__course_key=self.course_key,
+        ).values('answer_id', 'run', 'run__created').annotate(count=Count('answer_id'))
+        # Make sure every answer has a count and convert to JSON serializable format
+        response_counts = {(item['answer_id'], item['run']): item['count'] for item in response_data}
 
         problem = modulestore().get_item(self.wrapped_block_usage_key)
         tree = problem.lcp.tree
         choice_elements = tree.xpath('//choicegroup/choice')
         choices = [
             {
-                'answer_id': choice.get("name"),
+                'answer_id': choice.get('name'),
                 'answer_text': choice.text,
-                'count': response_counts.get(choice.get("name"), 0)
             } for choice in choice_elements
         ]
+        serialized_runs = [
+            {
+                'id': run.id,
+                'created': run.created.isoformat(),
+            } for run in runs
+        ]
+        counts = {
+            choice['answer_id']: {
+                run['id']: response_counts.get((choice['answer_id'], run['id']), 0)
+                for run in serialized_runs
+            } for choice in choices
+        }
```

Comment:
j/w: Are we structuring the data like this strictly for the benefit of D3? Intuitively I would have thought we'd return a structure like this:

```javascript
[
  {
    'run_id': 1,
    'created': '2018-03-29',
    'submissions': [
      {
        'answer_id': 'choice_0',
        'answer_text': 'text',
        'count': 5
      }
    ]
  }
]
```

Reply:
If there are no runs then that structure wouldn't return any answer data

Reply:
The only way that we can have any submissions is if there's an open run, right (https://github.com/mitodl/rapid-response-xblock/pull/37/files#diff-91cebc94c04b612041ae2af386ac2568R65). If that's right, then no runs = no submissions, and we don't have anything to show. I guess your current structure doesn't repeat the answer ids and text values for each run, so that's an advantage. I guess I'm sort of ambivalent. No need to change right now

Reply:
Even with no runs (before the button is clicked to open a new one) we still need to show the answers in the chart, on the x axis with empty bars